feature-libs/asm/core/store/actions/asm-ui.action.ts

```diff
@@ -4,8 +4,8 @@ import { AsmUi } from '../../models/asm.models';
 export const ASM_UI_UPDATE = '[Asm] UI Update';
 
 export class AsmUiUpdate implements Action {
-  readonly type = ASM_UI_UPDATE;
-  constructor(public payload: AsmUi) {}
+  readonly type: string = ASM_UI_UPDATE;
+  constructor(public payload?: AsmUi) {}
```

Comment:
Having to make the payload optional doesn't sound very good to me.  Is there an alternative?

Reply:
The fix to this is to use `Action` as the function parameter type in the reducer and then cast it to `AsmUiUpdate`